hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/validation/RequestFeatureValidator.java

```diff
@@ -96,4 +97,15 @@
    */
   Type requestType();
 
+
+  /**
+   * The max client version for which the validator would run. This condition happens in parallel with the specified
+   * validation conditions. The validator would run either one of the specified conditions apply or the request
+   * client version is equal to or older than the specified version. The value defaults to FUTURE_VERSION so the
+   * validator would never run for any of the clients, the validator could still run based on the specified
+   * conditions though.
+   * @returns the max required client version for which the validator runs for older clients.
+   */
+  ClientVersion maxClientVersion() default ClientVersion.FUTURE_VERSION;
```

Comment:
If we remove the `conditions` field then this probably shouldn't have a default value. Technically the request validators can be used for things other than client versioning but I don't think we have a use case for that right now. Narrowing the scope of conditions indicates what their intended use is.